src/App.tsx

```diff
@@ -1,27 +1,53 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const loadGoods = async (callback: Promise<Good[]>) => {
+    try {
+      const loadedGoods = await callback;
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+      setGoods(loadedGoods);
+    } catch (error) {
+      throw new Error('error');
+    }
+  };
 
-    <GoodsList goods={[]} />
-  </div>
-);
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
+
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={() => loadGoods(getAll())}
```

Comment:
Don't call the function here. Call it inside loadGoods
```suggestion
        onClick={() => loadGoods(getAll)}
```